README.md

```diff
@@ -0,0 +1,69 @@
+# OCP DNSNameResolver
+
+## Name
+
+OCP DNSNameResolver - updates the status of the [DNSNameResolver](https://github.com/openshift/api/blob/master/network/v1alpha1/types_dnsnameresolver.go)
+custom resources with IP addresses of matching resolved DNS names.
+
+## Description
+
+The spec of DNSNameResolver custom resource takes as input a DNS name. The DNS name can be either a regular or a wildcard DNS name. The plugin intercepts
+the DNS lookups of the matching DNS names and updates the status of these CRs with the IP addresses of the matching resolved names.
+
+The prerequisite for enabling this plugin are:
+- Adding the `DNSNameResolver` CRD to the Kubernetes API.
+- Adding `list` and `watch` permissions on the `DNSNameResolver` resources and `update` permission on the `DNSNameResolver/status` resource.
+
+NOTE: When adding the plugin to the `plugin.cfg` file in CoreDNS, care should be taken to place it before the plugins which will do the actual resolution of
+the DNS names that will be used in the DNSNameResolver custom resources. This will ensure that the plugin can intercept the DNS request and response in the
+plugin chain.
+
+## Syntax
+
+```
+ocp_dnsnameresolver {
+    [namespaces NAMESPACE..]
+    [minTTL MINTTL]
+    [failureThreshold FAILURE_THRESHOLD]
+}
+```
+
+- `namespaces` specifies those namespaces in which the DNSNameResolver custom resources will be monitored. When this option is omitted then DNSNameResolver
+custom resource of all namespaces will be monitored.
+- `minTTL` specifies the TTL value to be used for an IP address when the TTL in the DNS lookup response is zero OR when a DNS lookup fails and the TTL of the
+IP address has expired. If the option is omitted then the default value of 5 is used.
+- `failureThreshold` specifies the number of consecutive DNS lookup failures for a DNS name until the details of the DNS name can be removed from the status
+of a DNSNameResolver custom resource. However, the details of the DNS name will be removed only if the TTL of all the associated IP addresses have expired.
+If the option is omitted then the default value of 5 is used.
+
+## Examples
+
+Enabling the OCP DNSNameResolver plugin with all defaults and enabling monitoring of DNSNameResolver custom resources in all namespaces:
```

Comment:
```suggestion
Enabling the OCP DNSNameResolver plugin with all defaults:
```
You already mentioned the default value for `namespaces` option, I think it's fair to expect the reader to go up and check the defaults.

Reply:
Makes sense. Will update it.